src/microbe_stage/editor/PatchMapDrawer.cs

```diff
@@ -472,52 +504,183 @@ private void CreateRegionLinks()
 
-                connections.Add(connectionKey, pathToAdjacent);
+                connections.Add(connectionKey, (pathToAdjacent, startVertical, endVertical));
             }
         }
 
         AdjustPathEndpoints();
     }
 
+    private int AdjustPriority(int size1, int priority)
+    {
+        // Prioritize connections to the longer sides
+        priority += size1 * size1;
+
+        return priority;
+    }
+
+    private int GetIShapePathCharacteristics(int startRows, int startColumns, int endRows, int endColumns,
+        bool flip = false)
+    {
+        var priority = 5;
+
+        // Flip: false -> # of Columns, Flip: true -> # of Rows
+        priority = AdjustPriority(flip ? startRows : startColumns, priority);
+        priority = AdjustPriority(flip ? endRows : endColumns, priority);
+        return priority;
+    }
+
+    private int GetLShapePathCharacteristics(int startRows, int startColumns, int endRows,
+        int endColumns)
+    {
+        var priority = 3;
+        priority = AdjustPriority(startColumns, priority);
+        priority = AdjustPriority(endRows, priority);
+        return priority;
+    }
+
+    private (Vector2 Intermediate1, Vector2 Intermediate2, int Priority) GetZShapePathCharacteristics(int startRows,
+        int startColumns, int endRows, int endColumns, Vector2 startCenter, Vector2 endCenter,
+        Vector2 middlePoint, bool isTrueMiddle, bool flip = false)
+    {
+        Vector2 intermediate1;
+        Vector2 intermediate2;
+        int priority = isTrueMiddle ? 2 : 1;
+        if (!flip)
+        {
+            // Columns
+            intermediate1 = new Vector2(startCenter.X, middlePoint.Y);
+            intermediate2 = new Vector2(endCenter.X, middlePoint.Y);
+        }
+        else
+        {
+            // Rows
+            intermediate1 = new Vector2(middlePoint.X, startCenter.Y);
+            intermediate2 = new Vector2(middlePoint.X, endCenter.Y);
+        }
+
+        priority = AdjustPriority(flip ? startRows : startColumns, priority);
+        priority = AdjustPriority(flip ? endRows : endColumns, priority);
+        return (intermediate1, intermediate2, priority);
+    }
+
+    private (Vector2 Intermediate1, Vector2 Intermediate2, int Priority) GetUShapePathCharacteristics(int startRows,
+        int startColumns, int endRows, int endColumns, Rect2 startRect, Rect2 endRect, Vector2 startCenter,
+        Vector2 endCenter,
+        string shape, int i)
+    {
+        var upper = startRect.Position.Y < endRect.Position.Y ? startRect : endRect;
+        var lower = startRect.End.Y > endRect.End.Y ? startRect : endRect;
+        var left = startRect.Position.X < endRect.Position.X ? startRect : endRect;
+        var right = startRect.End.X > endRect.End.X ? startRect : endRect;
+        Vector2 intermediate1;
+        Vector2 intermediate2;
+        int priority = 0;
+        switch (shape)
+        {
+            case "lower":
```

Comment:
I think this should be an enum as using strings with magic values like this isn't very good style.
Also the CI has found unused variables so, could you please fix those before I take a full new look at the code?

Reply:
I updated the code